src/components/Viewport.ts

```diff
@@ -168,12 +170,25 @@ export default class Viewport {
     // Update position & nearestPanel
     state.position = pos;
     this.nearestPanel = this.findNearestPanel();
-
     const nearestPanel = this.nearestPanel;
     const originalNearestPosition = nearestPanel
       ? nearestPanel.getPosition()
       : 0;
 
+    // From 0(panel position) to 1(panel position + panel size)
+    // When it's on gap area value will be (val > 1 || val < 0)
+    if (nearestPanel) {
+      const hangerPosition = this.getHangerPosition();
+      const panelPosition = nearestPanel.getPosition();
+      const panelSize = nearestPanel.getSize();
+      const halfGap = options.gap / 2;
+
+      // As panel's range is from panel position - half gap ~ panel pos + panel size + half gap
+      state.relativePosition = (hangerPosition - panelPosition + halfGap) / (panelSize +  2 * halfGap);
```

Comment:
@WoodNeck 
The relativeHangerPosition, relativeAnchorPosition used for the state is known as just a position, but relativePosition here is the ratio of position. The name is vague.

And how about using getProgress?

Reply:
I agree name's very vague, but I don't like syntax of `getProgress`.
Putting same value twice looks very awful for me.
Like, `getProgress([hP, pP - hG, pP - hG, pP +pS + hG])`.
Also, it's a bit more efficient for a simple calculation like that.

Reply:
It meant using nearestPanel.getProgress. But I think the way it is now is better.